### ISSUE TITLE
Truncated network traffic in the Throughput test when the specified data size is greater than 0xFFFF [8574]

### ISSUE THREAD

**Issue by i-and** [eProsima/Fast-DDS#1237]
Apparently, the reason is to use the uint16_t as the constructor argument type:

~~~ c
typedef struct ThroughputType {
    uint32_t seqnum;
    std::vector<uint8_t> data;

    ThroughputType(
            uint16_t number)
        : seqnum(0)
        , data(number,0)
    {
    }
} ThroughputType;
~~~

**Comment by adolfomarver** [comment]
@i-and Try with #1070 it should solve the issue.

**Comment by EduPonz** [comment]
I'll go ahead and close this, since it was solved in #1070 . Please feel free to reopen at any time
